src/main/resources/static/js/testrules.js

```diff
@@ -221,14 +190,19 @@ jQuery(document).ready(
               sticky : false,
               theme : 'Notify'
             });
-            var eventsList = JSON.parse(fileContent);
-            ko.cleanNode($("#testEventsDOMObject")[0]);
-            vm.eventsBindingList.removeAll();
-            $('.eventsListDisplay > div:gt(0)').remove();
-            vm.eventsBindingList = ko.observableArray(eventsList);
-            ko.applyBindings(vm, $("#testEventsDOMObject")[0]);
-            closeTooltip();
-            loadTooltip();
+
+            var list = JSON.parse(fileContent);
+
+            if (type == true) {
+                vm.rulesBindingList([]);
+                vm.rulesBindingList(list);
+                vm.rulesBindingList.valueHasMutated();
```

Comment:
I'm not sure if valueHasMutated is even needed. It's used in subscription but try removing it and see if it affects anything.

Reply:
It is used to force the update of an observable, but it seems that it is not needed here. I removed it.